### ISSUE TITLE
much slower generation using autoGPTQ

### ISSUE THREAD

**Issue by TheHolyLoli** [oobabooga/text-generation-webui#2583]
### Describe the bug

using the latest commit in main branch,autogptq is much slower than gptq-for-llama  for both older models and new models quantized with autogptq.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

test autogptq and gptq-for-llama on the same 7b model

### Logs

autogptq:
```shell
INFO:Loading TheBloke_WizardLM-7B-uncensored-GPTQ...
INFO:The AutoGPTQ params are: {'model_basename': 'WizardLM-7B-uncensored-GPTQ-4bit-128g.compat.no-act-order', 'device': 'cuda:0', 'use_triton': False, 'use_safetensors': True, 'trust_remote_code': True, 'max_memory': None, 'quantize_config': BaseQuantizeConfig(bits=4, group_size=128, damp_percent=0.01, desc_act=False, sym=True, true_sequential=True, model_name_or_path=None, model_file_base_name=None)}
WARNING:The safetensors archive passed at models/TheBloke_WizardLM-7B-uncensored-GPTQ/WizardLM-7B-uncensored-GPTQ-4bit-128g.compat.no-act-order.safetensors does not contain metadata. Make sure to save your model with the `save_pretrained` method. Defaulting to 'pt' metadata.
WARNING:skip module injection for FusedLlamaMLPForQuantizedModel not support integrate without triton yet.
INFO:Loaded the model in 55.91 seconds.
Output generated in 30.31 seconds (1.09 tokens/s, 33 tokens, context 70, seed 1244650697)
```
gptq-for-llama:
```
INFO:Loading TheBloke_WizardLM-7B-uncensored-GPTQ...
INFO:Found the following quantized model: models/TheBloke_WizardLM-7B-uncensored-GPTQ/WizardLM-7B-uncensored-GPTQ-4bit-128g.compat.no-act-order.safetensors
INFO:Loaded the model in 2.45 seconds.
Output generated in 4.77 seconds (6.92 tokens/s, 33 tokens, context 70, seed 1868669003)
```

### System Info

- OS: fedora 37 KDE
 - Python: 3.11
 - AutoGPTQ: v0.2.2 installed from source 
 - GTX1070
 - NVIDIA-SMI 530.41.03 Driver Version: 530.41.03 CUDA Version: 12.1 (used CUDA version is 11.8)

**Comment by LoopControl** [comment]
I'm having the same problem - massive slowdown with autogptq.

When using GPTQ-for-llama, I'm averaging 15 tokens/s and with Autogptq, I'm seeing 2 tokens per second.

**Comment by tensiondriven** [comment]
Likley unrelated, but in case it is, https://github.com/oobabooga/text-generation-webui/issues/2602

**Comment by Ainaemaet** [comment]
Very slow for me as well. Spent the last 2 days trying everything I could think of to get it moving but it's abysmal.

**Comment by berkut1** [comment]
The same. It works x2 time slower, than the same model for CPU model.

**Comment by TheHolyLoli** [comment]
as stated in #2685 , one user seem to experience slow load time as well as slow generation

**Comment by kelheor** [comment]
I have normal load model time in both cases for 33B model:
Auto-GPTQ: 11.83 seconds.
GPTQ for LLAMA:  7.90 seconds.

**Performance comparison:**

**Auto-GPTQ:**

Output generated in 19.02 seconds (10.51 tokens/s, 200 tokens, context 43, seed 543723477)
Output generated in 28.24 seconds (7.08 tokens/s, 200 tokens, context 243, seed 1817225413)
Output generated in 37.63 seconds (5.32 tokens/s, 200 tokens, context 443, seed 922513496)
Output generated in 53.51 seconds (3.74 tokens/s, 200 tokens, context 643, seed 1321907924)
Output generated in 68.00 seconds (2.94 tokens/s, 200 tokens, context 842, seed 1382678522)
Output generated in 79.75 seconds (2.51 tokens/s, 200 tokens, context 1042, seed 393716294)
Output generated in 90.04 seconds (2.22 tokens/s, 200 tokens, context 1240, seed 41524587)
Output generated in 60.19 seconds (0.63 tokens/s, 38 tokens, context 1440, seed 1582526936)
Output generated in 49.39 seconds (0.02 tokens/s, 1 tokens, context 1477, seed 1216560646)

**GPTQ for LLAMA:**

Output generated in 22.93 seconds (8.72 tokens/s, 200 tokens, context 43, seed 221987110)
Output generated in 23.24 seconds (8.60 tokens/s, 200 tokens, context 243, seed 1291998717)
Output generated in 23.37 seconds (8.56 tokens/s, 200 tokens, context 443, seed 1015859689)
Output generated in 23.51 seconds (8.51 tokens/s, 200 tokens, context 643, seed 378698710)
Output generated in 23.65 seconds (8.45 tokens/s, 200 tokens, context 843, seed 57608041)
Output generated in 23.85 seconds (8.39 tokens/s, 200 tokens, context 1043, seed 1726965209)
Output generated in 36.10 seconds (5.54 tokens/s, 200 tokens, context 1243, seed 573124894)
Output generated in 35.91 seconds (5.57 tokens/s, 200 tokens, context 1443, seed 146068407)
Output generated in 45.21 seconds (4.42 tokens/s, 200 tokens, context 1643, seed 593350328)
Output generated in 9.48 seconds (3.06 tokens/s, 29 tokens, context 1843, seed 2013420097)

As you can see, there is a huge performance drop with Auto-GPTQ over time.

**Comment by LoopControl** [comment]
I've found the fix!

There is a `use_cuda_fp16` parameter for the model load in autoGPTQ. Setting it to "False" takes my tokens/s from 2-3 tok/s to 15-16 tok/s which matches what GPTQ-for-llama does!

On the current commit, in `AutoGPTQ_loader.py`, find the section that stats with:

```python
    # Define the params for AutoGPTQForCausalLM.from_quantized
    params = {
```

And add ''use_cuda_fp16': False,` in that `params` block - like this:
```python
    params = {
        'use_cuda_fp16': False,
        'model_basename': pt_path.stem,
        'device': "cuda:0" if not shared.args.cpu else "cpu",
        'use_triton': shared.args.triton,
        'inject_fused_attention': not shared.args.no_inject_fused_attention,
        'inject_fused_mlp': not shared.args.no_inject_fused_mlp,
        'use_safetensors': use_safetensors,
        'trust_remote_code': shared.args.trust_remote_code,
        'max_memory': get_max_memory_dict(),
        'quantize_config': quantize_config,
    }
```

@oobabooga - Please take a look and see if this is safe to add. It makes Autogptq run at same speed as gptq-for-llama for me.

**Comment by TheHolyLoli** [comment]
@oobabooga  can confirm this fixes Autogptq being much slower than  gptq-for-llama on my system.this actually makes it faster and uses less vram

**Comment by oobabooga** [comment]
I have added this as a parameter here https://github.com/oobabooga/text-generation-webui/commit/3ae9af01aa9fab164a9cf85f0309ee7009f25477

For me it makes models slightly slower.

**Comment by TheHolyLoli** [comment]
@LoopControl @kelheor are you using 10th-gen Nvidia gpus(non rtx)?
1070, 1080,...?

**Comment by LoopControl** [comment]
@TheHolyLoli Yes, I run my models on a (non-RTX) Tesla P40 which is sm_61 compute capability.

@oobabooga Looks great, thanks for adding a CLI flag for that!

**Comment by FartyPants** [comment]
With or without the flag it's basically at worse speed than CPU on my 3090.
No indication that CUDA is not used though.

Any idea how to rebuild the wheel on my system directly?
